src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferStreamAsyncMemoryOverloads.Fixer.cs

```diff
@@ -0,0 +1,112 @@
+            {
+                return;
+            }
+
+            SemanticModel model = await doc.GetSemanticModelAsync(ct).ConfigureAwait(false);
+
+            if (!(model.GetOperation(node, ct) is IInvocationOperation invocation))
+            {
+                return;
+            }
+
+            // Defensive check to ensure the fix is only attempted on one of the 4 specific undesired overloads
+            if (invocation.Arguments.Length < 3 || invocation.Arguments.Length > 4)
+            {
+                return;
+            }
+
+            string title = MicrosoftNetCoreAnalyzersResources.PreferStreamAsyncMemoryOverloadsTitle;
+
+            context.RegisterCodeFix(
+                CodeAction.Create(
+                    title: title,
+                    createChangedDocument: c => FixInvocation(doc, root, invocation, invocation.TargetMethod.Name),
+                    equivalenceKey: title),
+                context.Diagnostics);
+        }
+
+        private static Task<Document> FixInvocation(Document doc, SyntaxNode root, IInvocationOperation invocation, string methodName)
+        {
+            SyntaxGenerator generator = SyntaxGenerator.GetGenerator(doc);
+
+            // The stream object
+            SyntaxNode instanceNode = invocation.Instance.Syntax;
+
+            // Need the byte array object so we can invoke its AsMemory() method
+            SyntaxNode bufferInstanceNode = invocation.Arguments[0].Value.Syntax; // byte[] buffer
+
+            // These arguments are not modified, just moved inside AsMemory
+            SyntaxNode offsetNode = invocation.Arguments[1].Syntax; // int offset
+            SyntaxNode countNode = invocation.Arguments[2].Syntax;  // int count
+
+            // Generate an invocation of the AsMemory() method from the byte array object
+            SyntaxNode asMemoryExpressionNode = generator.MemberAccessExpression(bufferInstanceNode, "AsMemory");
+            SyntaxNode asMemoryInvocationNode = generator.InvocationExpression(asMemoryExpressionNode, offsetNode, countNode);
+
+            // Create a new async method call for the stream object, no arguments yet
+            SyntaxNode asyncMethodNode = generator.MemberAccessExpression(instanceNode, methodName);
+
+            // Add the arguments to the async method call, with or without CancellationToken
+            SyntaxNode newInvocationExpression;
+            if (invocation.Arguments.Length == 4)
+            {
+                newInvocationExpression = generator.InvocationExpression(
+                    asyncMethodNode, asMemoryInvocationNode,
+                    invocation.Arguments[3].Syntax /* CancellationToken */);
+            }
+            else
+            {
+                newInvocationExpression = generator.InvocationExpression(asyncMethodNode, asMemoryInvocationNode);
+            }
+
+            SyntaxNode newRoot = generator.ReplaceNode(root, invocation.Syntax, newInvocationExpression);
```

Comment:
I should also include the trivia.

Reply:
Yes, and please do add unit test for the trivia case!

Reply:
Will do!

Reply:
Still pending to address trivia. The latest commits were just for the named arguments.

Reply:
Trivia handling added. Please take a look.